backend/src/ml_space_lambda/utils/exceptions.py

```diff
@@ -24,3 +24,14 @@ def __init__(self, message, http_status_code):
 class ResourceNotFound(ServiceException):
     def __init__(self, message):
         super().__init__(message, 404)
+
+class ResourceInUseError(ServiceException):
+    """
+    Thrown when attempting to delete or modify a resource that is currently in use.
+
+    Attributes:
+        message: error description
+    """
+    def __init__(self, message: str = None):
+        default = "Resource is currently in use and cannot be deleted."
+        super().__init__(message or default, 409)
```

Comment:
Move the default out to a class constant and make it clear in the method signature that is the default.

```suggestion
class ResourceInUseError(ServiceException):
    DEFAULT_MESSAGE = "Resource is currently in use and cannot be deleted."
    """
    Thrown when attempting to delete or modify a resource that is currently in use.

    Attributes:
        message: error description
    """
    def __init__(self, message: str = DEFAULT_MESSAGE):
        super().__init__(message, 409)
```